File: mobile/android/java/TigerExAuth.java
/**
 * TigerEx Android Authentication (Java)
 * @file TigerExAuth.java
 * @description Authentication for Android native Java apps
 * @author TigerEx Development Team
 */

package com.tigerex.auth;

import android.content.Context;
import android.content.SharedPreferences;
import org.json.JSONObject;
import java.util.concurrent.TimeUnit;

/**
 * TigerEx Authentication Manager for Android (Java)
 * 
 * Usage:
 * - Login: TigerExAuth.login(context, email, name)
 * - Logout: TigerExAuth.logout(context)
 * - Check: TigerExAuth.isLoggedIn(context)
 */
public class TigerExAuth {
    
    private static final String PREFS_NAME = "tigerex_auth";
    private static final String KEY_TOKEN = "token";
    private static final String KEY_USER = "user";
    private static final String KEY_EXPIRY = "expiry";
    
    /**
     * Get SharedPreferences
     */
    private static SharedPreferences getPrefs(Context context) {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
    }
    
    /**
     * Check if user is logged in
     */
    public static boolean isLoggedIn(Context context) {
        SharedPreferences prefs = getPrefs(context);
        String token = prefs.getString(KEY_TOKEN, null);
        long expiry = prefs.getLong(KEY_EXPIRY, 0);
        
        if (token == null || token.isEmpty()) return false;
        
        if (expiry > 0 && expiry < System.currentTimeMillis()) {
            logout(context);
            return false;
        }
        
        return true;
    }
    
    /**
     * Get current user data
     */
    public static User getUser(Context context) {
        SharedPreferences prefs = getPrefs(context);
        String userJson = prefs.getString(KEY_USER, null);
        
        if (userJson == null) return null;
        
        try {
            JSONObject json = new JSONObject(userJson);
            return new User(
                json.getString("email"),
                json.optString("name", null)
            );
        } catch (Exception e) {
            return null;
        }
    }
    
    /**
     * Get user email
     */
    public static String getEmail(Context context) {
        User user = getUser(context);
        return user != null ? user.getEmail() : "";
    }
    
    /**
     * Get display name
     */
    public static String getDisplayName(Context context) {
        User user = getUser(context);
        if (user != null && user.getName() != null) {
            return user.getName();
        }
        String email = getEmail(context);
        int atIndex = email.indexOf('@');
        return atIndex > 0 ? email.substring(0, atIndex) : "User";
    }
    
    /**
     * Get avatar initial
     */
    public static String getAvatar(Context context) {
        String name = getDisplayName(context);
        return name.isEmpty() ? "U" : String.valueOf(name.charAt(0)).toUpperCase();
    }
    
    /**
     * Login user
     * @return true if successful
     */
    public static boolean login(Context context, String email, String name) {
        if (email == null || email.isEmpty()) return false;
        
        SharedPreferences prefs = getPrefs(context);
        String token = "tigerex_token_" + System.currentTimeMillis();
        long expiry = System.currentTimeMillis() + TimeUnit.HOURS.toMillis(24);
        
        JSONObject userJson = new JSONObject();
        try {
            userJson.put("email", email);
            if (name != null && !name.isEmpty()) {
                userJson.put("name", name);
            }
        } catch (Exception e) {
            return false;
        }
        
        SharedPreferences.Editor editor = prefs.edit();
        editor.putString(KEY_TOKEN, token);
        editor.putString(KEY_USER, userJson.toString());
        editor.putLong(KEY_EXPIRY, expiry);
        editor.apply();
        
        return true;
    }
    
    /**
     * Logout user
     */
    public static void logout(Context context) {
        SharedPreferences prefs = getPrefs(context);
        prefs.edit().clear().apply();
    }
    
    /**
     * User data class
     */
    public static class User {
        private final String email;
        private final String name;
        
        public User(String email, String name) {
            this.email = email;
            this.name = name;
        }
        
        public String getEmail() { return email; }
        public String getName() { return name; }
    }
}

/**
 * Base Activity with Auth
 */
class TigerExBaseActivity extends androidx.appcompat.app.AppCompatActivity {
    
    protected boolean isLoggedIn() {
        return TigerExAuth.isLoggedIn(this);
    }
    
    protected TigerExAuth.User getUser() {
        return TigerExAuth.getUser(this);
    }
    
    protected boolean login(String email, String name) {
        return TigerExAuth.login(this, email, name);
    }
    
    protected void logout() {
        TigerExAuth.logout(this);
    }
}

/**
 * Base Fragment with Auth
 */
class TigerExBaseFragment extends androidx.fragment.app.Fragment {
    
    protected boolean isLoggedIn() {
        Context context = getContext();
        return context != null && TigerExAuth.isLoggedIn(context);
    }
    
    protected TigerExAuth.User getUser() {
        Context context = getContext();
        return context != null ? TigerExAuth.getUser(context) : null;
    }
    
    protected boolean login(String email, String name) {
        Context context = getContext();
        return context != null && TigerExAuth.login(context, email, name);
    }
    
    protected void logout() {
        Context context = getContext();
        if (context != null) {
            TigerExAuth.logout(context);
        }
    }
}public static Wallet createWallet() {
    String chars = "0123456789abcdef";
    String addr = "0x";
    for(int i=0;i<40;i++) addr += chars.charAt((int)(Math.random()*16));
    String seed = "abandon ability able about above absent absorb abstract absurd abuse access accident account accuse achieve acid acoustic acquire across act action actor actress actual adapt add adjust admin admit adult advance advice aerobic affair afford afraid again age agency agent agree ahead aim air airport alarm album alcohol alien alike alive allow alone along alpha already also alter always amazing among amount analyze ancient angle angry animal anniversary announce another answer antenna anxiety any apart apology appear apple approve april aqua arabian architecture area argue arise armed armor army around arrange arrest arrival arrive arrow artist artwork area";
    return new Wallet(addr, seed, "USER_OWNS");
}

File: mobile/android/kotlin/TigerExAuth.kt
/**
 * TigerEx Android Authentication (Kotlin)
 * @file TigerExAuth.kt
 * @description Authentication for Android native apps
 * @author TigerEx Development Team
 */

package com.tigerex.auth

import android.content.Context
import android.content.SharedPreferences
import org.json.JSONObject
import java.util.concurrent.TimeUnit

/**
 * TigerEx Authentication Manager for Android
 * 
 * Usage:
 * - Login: TigerExAuth.login(context, email, name)
 * - Logout: TigerExAuth.logout(context)
 * - Check: TigerExAuth.isLoggedIn(context)
 */
object TigerExAuth {
    
    private const val PREFS_NAME = "tigerex_auth"
    private const val KEY_TOKEN = "token"
    private const val KEY_USER = "user"
    private const val KEY_EXPIRY = "expiry"
    
    /**
     * SharedPreferences instance
     */
    private fun getPrefs(context: Context): SharedPreferences {
        return context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }
    
    /**
     * Check if user is logged in
     */
    fun isLoggedIn(context: Context): Boolean {
        val prefs = getPrefs(context)
        val token = prefs.getString(KEY_TOKEN, null)
        val expiry = prefs.getLong(KEY_EXPIRY, 0)
        
        if (token.isNullOrEmpty()) return false
        if (expiry > 0 && expiry < System.currentTimeMillis()) {
            logout(context)
            return false
        }
        
        return true
    }
    
    /**
     * Get current user data
     */
    fun getUser(context: Context): User? {
        val prefs = getPrefs(context)
        val userJson = prefs.getString(KEY_USER, null) ?: return null
        
        return try {
            val json = JSONObject(userJson)
            User(
                email = json.getString("email"),
                name = json.optString("name", null)
            )
        } catch (e: Exception) {
            null
        }
    }
    
    /**
     * Get user email
     */
    fun getEmail(context: Context): String {
        return getUser(context)?.email ?: ""
    }
    
    /**
     * Get display name
     */
    fun getDisplayName(context: Context): String {
        val user = getUser(context)
        return user?.name ?: user?.email?.split("@")?.get(0) ?: "User"
    }
    
    /**
     * Get avatar initial
     */
    fun getAvatar(context: Context): String {
        return getDisplayName(context).firstOrNull()?.uppercaseChar() ?: 'U'
    }
    
    /**
     * Login user
     * @return true if successful
     */
    fun login(context: Context, email: String, name: String? = null): Boolean {
        if (email.isBlank()) return false
        
        val prefs = getPrefs(context)
        val token = "tigerex_token_${System.currentTimeMillis()}"
        val expiry = System.currentTimeMillis() + TimeUnit.HOURS.toMillis(24)
        
        val userJson = JSONObject().apply {
            put("email", email)
            name?.let { put("name", it) }
        }
        
        prefs.edit().apply {
            putString(KEY_TOKEN, token)
            putString(KEY_USER, userJson.toString())
            putLong(KEY_EXPIRY, expiry)
            apply()
        }
        
        return true
    }
    
    /**
     * Logout user
     */
    fun logout(context: Context) {
        val prefs = getPrefs(context)
        prefs.edit().clear().apply()
    }
    
    /**
     * User data class
     */
    data class User(
        val email: String,
        val name: String? = null
    )
}

/**
 * Auth State Listener
 */
interface AuthStateListener {
    fun onAuthStateChanged(isLoggedIn: Boolean)
}

/**
 * Base Activity with Auth
 */
open class TigerExActivity : androidx.appcompat.app.AppCompatActivity() {
    
    private val authListeners = mutableListOf<AuthStateListener>()
    
    protected fun isLoggedIn(): Boolean = TigerExAuth.isLoggedIn(this)
    
    protected fun getUser(): TigerExAuth.User? = TigerExAuth.getUser(this)
    
    protected fun login(email: String, name: String? = null): Boolean = TigerExAuth.login(this, email, name)
    
    protected fun logout() {
        TigerExAuth.logout(this)
        notifyAuthStateChanged(false)
    }
    
    protected fun addAuthListener(listener: AuthStateListener) {
        authListeners.add(listener)
    }
    
    protected fun removeAuthListener(listener: AuthStateListener) {
        authListeners.remove(listener)
    }
    
    private fun notifyAuthStateChanged(isLoggedIn: Boolean) {
        authListeners.forEach { it.onAuthStateChanged(isLoggedIn) }
    }
}

/**
 * Base Fragment with Auth
 */
open class TigerExFragment : androidx.fragment.app.Fragment() {
    
    protected fun isLoggedIn(): Boolean = 
        context?.let { TigerExAuth.isLoggedIn(it) } ?: false
    
    protected fun getUser(): TigerExAuth.User? = 
        context?.let { TigerExAuth.getUser(it) }
    
    protected fun login(email: String, name: String? = null): Boolean = 
        context?.let { TigerExAuth.login(it, email, name) } ?: false
    
    protected fun logout() {
        context?.let { TigerExAuth.logout(it) }
    }
}